bin/pycbc_live

```diff
@@ -514,14 +515,21 @@ class LiveEventManager(object):
         if not upload_checks:
             event.save(fname)
         if upload_checks or optimize_snr_checks:
+            live_ifos = [ifo for ifo in sld if 'snr_series' in sld[ifo]]
             if optimize_snr_checks:
                 logging.info('Optimizing SNR for coinc above threshold ..')
                 # Tell snr optimized event about p_terr
                 if hasattr(event, 'p_terr') and event.p_terr is not None:
                     coinc_results['p_terr'] = event.p_terr
-            live_ifos = [ifo for ifo in sld if 'snr_series' in sld[ifo]]
+                # Run the setup to gather/save all the data before threading
+                cmd, out_dir_path = self.setup_optimize_snr(
+                    coinc_results,
+                    live_ifos,
+                    coinc_ifos,
+                    fname,
+                )
             thread_args = (
-                event, fname, comment, coinc_results, live_ifos, coinc_ifos,
+                event, fname, comment, cmd, out_dir_path
```

Comment:
```suggestion
                event, fname, comment, cmd, out_dir_path,
```

Reply:
Ah, yes, sorry, I fixed that in my test but forgot to commit again.

Reply:
Ok fixed, and rebased etc. Always a wild ride for me.